### PR TITLE
allow browserify

### PR DESCRIPTION
**Pull request by t128**
Hi,

you can bundle node packages with browserify. Unfortunately it only works with static requires. So i changed all *require-all* with static requires.

### CODE REVIEW

**Comment by sidorares** [comment]
@t128 what is the point of browserifying node-mysql?

**Comment by t128** [comment]
bundle it into one js file eg: https://github.com/substack/node-browserify/issues/1277#issuecomment-115198436

**Comment by sidorares** [comment]
ok, it might make sense with node-webkit, but why bundle there when you can use native require?

**Comment by sidorares** [comment]
also see related #961

**Comment by sidorares** [comment]
sorry, have not read your comment until the end. ~50mb to ~1mb seems like a lot, but I guess it's not because of one file but because you exclude unused files. This can be done via coverage-like instrumentation to mark files for exclusion before packaging

**Comment by t128** [comment]
yeah, sorry, i changed the comment.

so, it will never happen, as only node.js is really supported?

ps: uglify halved the size to 400kb :)

**Comment by sidorares** [comment]
I'm not completely against your PR, but forcing only static require is actually bad for performance: I often require conditionally features that not always used to make startup time better

**Comment by dougwilson** [comment]
We would only even accept this PR if this library was usable in a browser, but as far as I know, you cannot use it in a browser because you cannot make raw sockets there.

Could you perhaps show/tell us how you are using this module in a web browser?

**Comment by sidorares** [comment]
@dougwilson it's used in node-webkit and bundled for performance considerations, not because of lack of module system

File: lib/protocol/packets/index.js [hunk 1]
```diff
@@ -1,4 +1,23 @@
...
+  ComQueryPacket: require('./ComQueryPacket'),
+  ComQuitPacket: require('./ComQuitPacket'),
+  ComStatisticsPacket: require('./ComStatisticsPacket'),
+  EmptyPacket: require('./EmptyPacket'),
+  EofPacket: require('./EofPacket'),
+  ErrorPacket: require('./ErrorPacket'),
+  Field: require('./Field'),
+  FieldPacket: require('./FieldPacket'),
+  HandshakeInitializationPacket: require('./HandshakeInitializationPacket'),
+  index: require('./index'),
```
Comment (dougwilson) [inline lib/protocol/packets/index.js]: Why are you requiring itself? You can remove this.

File: lib/protocol/sequences/index.js [hunk 2]
```diff
@@ -1,4 +1,10 @@
-var Elements = module.exports = require('require-all')({
-  dirname : __dirname,
-  filter  : /([A-Z].+)\.js$/,
-});
+module.exports = {
+  ChangeUser: require('./ChangeUser'),
+  Handshake: require('./Handshake'),
+  index: require('./index'),
```
Comment (dougwilson) [inline lib/protocol/sequences/index.js]: Why are you requiring itself? You can remove this.

**Comment by dougwilson** [comment]
Oh, gotcha. I left a couple line notes of two things that need to be fixed.

**Comment by t128** [comment]
hmm, you'r right. i missed the *[A-Z]*

**Comment by dougwilson** [comment]
We can probably remove `require-all` from the list of dependencies now, no?

**Comment by t128** [comment]
no, not really: https://github.com/felixge/node-mysql/blob/de5913227dbbaacbbdf22ee38689d090e1451be9/test/run-cov.js#L29

**Comment by sidorares** [comment]
move it to devDependencies ?

**Comment by t128** [comment]
dat error:
```
AssertionError: 'Fri Jun 26 2015 09:50:00 GMT+0000 (Coordinated Universal Time)' === 'Fri Jun 26 2015 09:49:00 GMT+0000 (Coordinated Universal Time)'
```
:(

**Comment by dougwilson** [comment]
@t128 don't worry about that error :) We had a volunteer who would look into it at #1045 but still no resolution.

**Pull request closed without merging**